SonarQube.Client/RoslynExporterAdapter/RoslynRuleSetGenerator.cs

```diff
@@ -0,0 +1,157 @@
+﻿/*
+ * SonarQube Client
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using SonarQube.Client.Models;
+
+// Copied from the SonarScanner for MSBuild.
+// See https://github.com/SonarSource/sonar-scanner-msbuild/blob/9ccfdb648a0411014b29c7aee8e347aeab87ea71/src/SonarScanner.MSBuild.PreProcessor/Roslyn/RoslynRuleSetGenerator.cs#L28
+
+namespace SonarQube.Client.RoslynExporterAdapter
+{
+    public class RoslynRuleSetGenerator
+    {
+        private const string SONARANALYZER_PARTIAL_REPO_KEY = "sonaranalyzer-{0}";
+        private const string ROSLYN_REPOSITORY_PREFIX = "roslyn.";
+
+        private readonly IDictionary<string, string> sonarProperties;
+        private readonly string inactiveRuleActionText = GetActionText(RuleAction.None);
+
+        private string activeRuleActionText = GetActionText(RuleAction.Warning);
```

Comment:
Currently all active rules are returned as warnings. There's a specific SLVS ticket about that so I'll address it later. At the moment I'm just trying to remove the calls to the profile exporter API.